### PR TITLE
Features/widget test amountcard

### PR DESCRIPTION
**Pull request by krushnakantanayak**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-64

### Description
-Valid Balance Amount: Ensures correct balance amount and label are displayed.
-Large Balance Amount: Verifies that very large amounts are truncated and display an ellipsis.
-Empty Balance Amount: Confirms correct rendering when no amount is provided.

- 

### Screenshots

Amount_card_test

### Tips

Any tips for testing it?

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [ ] All items in the author checklist is checked or not applicable
- [ ] All files including .lock and assets are reviewed
- [ ] Reasonable UI flows/cases related to the change are manually tested
- [ ] No commented-out code or unnecessary debug statements.
- [ ] No spelling or grammatical errors in code comments or documentation.
- [ ] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [ ] The branch is up-to-date with the target branch (e.g., staging).
- [ ] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

File: test/widgets/balance_amount_test.dart [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        'Then it should display the correct balance amount and label text',
+        (WidgetTester tester) async {
+      // Arrange
+      const testBalance = '10000';
+
+      // Act
+      await tester.pumpWidget(createWidgetUnderTest(testBalance));
+
+      // Assert
+      expect(find.text('10000'), findsOneWidget);
```
Comment (quiteachaos) [inline test/widgets/balance_amount_test.dart]: expect 10,000
it should be formatted
change the balance amount file if req.

File: test/widgets/balance_amount_test.dart [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      final textFinder = find.text(largeBalance);
+      final textWidget = tester.widget<Text>(textFinder);
+      expect(textWidget.maxLines, 2);
+      expect(textWidget.overflow, TextOverflow.ellipsis);
+    });
+
+    testWidgets(
+        'Given an empty balance amount '
+        'when BalanceAmount is rendered '
+        'Then it should display an empty space and the label text',
```
Comment (quiteachaos) [inline test/widgets/balance_amount_test.dart]: Then it should display zero with rupee symbol and label text

update the test case here accordingly and the original file if req.

**Review by quiteachaos — CHANGES_REQUESTED** [review]

File: test/widgets/balance_amount_test.dart (hunk 1 above)
Comment (krushnakantanayak) [reply test/widgets/balance_amount_test.dart]: resolve the issue

File: test/widgets/balance_amount_test.dart (hunk 2 above)
Comment (krushnakantanayak) [reply test/widgets/balance_amount_test.dart]: updated

**Review by quiteachaos — APPROVED** [review]

**Pull request merged**